### PR TITLE
Allow group backend to declare users as admins

### LINKED ISSUE #5570: Allow group backends to define admin users
> As a developer of an authorization backend having a group concept in an Atlassian Crowd directory I want be able to follow the design already implemented in other applications like Jira or Confluence and define a `nextcloud-admins` group inside Crowd. To make them take effect this would require my `GroupInterface` to have the chance to answer the `isAdmin` question.

## Current situation:

```php
	/**
	 * Checks if a userId is in the admin group
	 * @param string $userId
	 * @return bool if admin
	 */
	public function isAdmin($userId) {
		return $this->isInGroup($userId, 'admin');
	}
```
([lib/private/Group/Manager.php#L285-L292](https://github.com/nextcloud/server/blob/master/lib/private/Group/Manager.php#L285-L292))

The `isAdmin` function is coded hard into the `\OC\Group\Manager` instead of delegated to a registered backend.

## Expected situation

The backend is able to define a `\OCP\GroupInterface::IS_ADMIN` action / capability and will then get asked for `isAdmin($userId)`. If the backend does answer `false` or does not have the capability, the current mechanic is used as a fallback.

### PR DESCRIPTION
**Pull request by Luzifer**
Implementation suggested in issue #5570 

fixes #5570

### CODE REVIEW

File: lib/public/GroupInterface.php [hunk 1]
```diff
@@ -51,6 +51,7 @@
 	//OBSOLETE const GET_DISPLAYNAME	= 0x00010000;
 	const COUNT_USERS		= 0x00100000;
 	const GROUP_DETAILS		= 0x01000000;
+	const IS_ADMIN  = 0x10000000;
```
Comment (MorrisJobke) [inline lib/public/GroupInterface.php]: Could you add a `@since` tag to the constant? 13.0.0 it will be.

**Comment by MorrisJobke** [comment]
cc @rullzer @nickvergessen @blizzz @schiessle @icewind1991

**Comment by Luzifer** [comment]
Added the tag. Hope it's how you expected it…

**Comment by MorrisJobke** [comment]
> Added the tag. Hope it's how you expected it…

Nice 👍

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
Do you have a backend that is implementing it?

**Review by icewind1991 — APPROVED** [review]

**Comment by Luzifer** [comment]
Not yet but as soon as it's available I will modify my backend… The backend is not (yet) public available… I'm using it to have the Nextcloud login delegated to an Atlassian Crowd instance… For all other applications connected the administrators are managed through Crowd, only for Nextcloud I need to set the admins inside the application instead of inside Crowd…

**Pull request merged**